tests/apps/core/test_core_api_room_user_access.py

```diff
@@ -163,171 +358,504 @@ def test_api_room_user_accesses_create_anonymous(self):
         self.assertEqual(
             response.json(),
-            {"room": ["You must be administrator of a room to add accesses to it."]},
+            {
+                "detail": "You must be administrator or owner of a room to add accesses to it."
+            },
         )
-        self.assertFalse(RoomUserAccess.objects.exists())
+        self.assertFalse(RoomUserAccess.objects.filter(user=other_user).exists())
 
-    def test_api_room_user_accesses_create_administrator_users(self):
+    def test_api_room_user_accesses_create_members(self):
         """
-        Direct administrators of a room should be allowed to create a room/user access
-        in this room.
+        A user who is a simple member in a room should not be allowed to create
+        room user accesses in this room.
         """
-        user = UserFactory()
-        room = RoomFactory(users=[(user, True)])
-        other_user = UserFactory()
-        is_administrator = random.choice([True, False])
+        user, other_user = UserFactory.create_batch(2)
+        group = GroupFactory(members=[user])
+        room = RoomFactory(users=[(user, "member")], groups=[(group, "member")])
 
         jwt_token = AccessToken.for_user(user)
 
         response = self.client.post(
-            f"/api/rooms/{room.id!s}/users/",
+            "/api/room-user-accesses/",
             {
                 "user": str(other_user.id),
-                "is_administrator": is_administrator,
+                "room": str(room.id),
+                "role": random.choice(["member", "administrator", "owner"]),
             },
             HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
 
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {
+                "detail": "You must be administrator or owner of a room to add accesses to it."
+            },
+        )
+        self.assertFalse(RoomUserAccess.objects.filter(user=other_user).exists())
+
+    def test_api_room_user_accesses_create_administrators_except_owner_direct(self):
+        """
+        A user who is direct administrator in a room should be allowed to create
+        room user accesses in this room for roles other than owner.
+        """
+        user, other_user = UserFactory.create_batch(2)
+        room = RoomFactory(users=[(user, "administrator")])
+
+        jwt_token = AccessToken.for_user(user)
+
+        response = self.client.post(
+            "/api/room-user-accesses/",
+            {
+                "user": str(other_user.id),
+                "room": str(room.id),
+                "role": random.choice(["member", "administrator"]),
+            },
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
         self.assertEqual(response.status_code, 201)
         self.assertEqual(RoomUserAccess.objects.count(), 2)
         self.assertTrue(RoomUserAccess.objects.filter(user=other_user).exists())
 
```

Comment:
```suggestion
 response = self.client.post(
            "/api/room-user-accesses/",
            {
                "user": str(other_user.id),
                "room": str(room.id),
                "role": "owner",
            },
            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
        )
        self.assertEqual(response.status_code, 403)
        self.assertEqual({"detail":"Only owners of a room can assign other users as owners."}, response.json())
```

Reply:
I didn't get this comment.

Reply:
ok understood. This was already handled in a subsequent test.